### PR TITLE
Support new booster notation (EXPOSUREAPP-11272)

### PR DESCRIPTION
**Pull request by jurajkusnier**
### Testing
- scan as many DCC QR codes as you can and carefully watch which certificate has been selected as the certificate with the highest priority.
- focus on vaccination certificates where `doseNumber` > `totalSeriesOfDoses` or `v[0].dn is > than v[0].sd`
- you can generate such certificates using cli tool, eg:
```
cwa vc gen -o --pii 32 -s v.0.dt=2021-08-01T08:57:25Z -s v.0.dn=2 -s v.0.sd=1
```

### Jira Link
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11272

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt [hunk 1]
```diff
@@ -54,14 +54,14 @@ fun Collection<CwaCovidCertificate>.toCertificateSortOrder(): List<CwaCovidCerti
...
- * Find Vaccination Certificates where total number of doses == number of administered doses and
- * 3.1 For vaccines with dose 3/3, priority will be received right away
- * 3.2 For BioNTech/Moderna/AstraZeneca vaccines that are taken after a recovery, priority will be received right away
- * 3.3 For J&J vaccines with dose 2/2, priority will be received right away
- * 3.4 If none of the criteria above is met, priority will be received after a 14 day period
- * If there is one or more certificates matching these requirements,
- * the first one is returned as a result of the operation.
+ * find Vaccination Certificates (i.e. DGC with v[0]) where v[0].dn is >= than v[0].sd and one of
+ * the time difference between the time represented by v[0].dt and the current device time is > 14 days, or
+ * v[0].dn is > 1 and v[0].mp equals EU/1/20/1525 (Johnson & Johnson) (booster vaccination), or
```
Comment (vaubaehn) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt:59]: Hi @jurajkusnier ,
> v[0].dn is > 1 and v[0].mp equals EU/1/20/1525 (Johnson & Johnson) (booster vaccination)

For my understanding that won't work, when v[0] for both conditions is the _same_ DCC. I assume the array v[0] refers to the vaccination entry of _any_ DCC.
Think, a user was J&J-jabbed and enters that certificate into CWA first. Then v[0][1].dn is == 1 and v[0][1].mp is == EU/1/20/1525. Then user gets the booster with mRNA and enters the certificate as v[0][2]. Now v[0][2].dn is > 1and v[0][2].mp is different than EU/1/20/1525.
So, the check/filter would need to process 2 different certificates for that single result.

**Review by vaubaehn — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt (hunk 1 above)
Comment (vaubaehn) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt:59]: Sorry, please ignore my above comment.
I now understood that above condition refers to a booster _with_ J&J, not a booster _after_ J&J.
Then `v[0].dn is > than v[0].sd` will do the trick for the new EU notation and boosters _after_ J&J or recovery.
Sorry again

**Review by SamuraiKek — APPROVED** [review]
LGTM. Tested with the scenarios from #4343 and adjusted the 2/2 J&J vaccines to 2/1. Everything looks alright.

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Pull request merged**